### ISSUE TITLE
about running under Windows

### ISSUE THREAD

**Issue by ez4encef** [megvii-research/CADDM#26]
Hello, can your code run under Windows? I have successfully run your code under Ubuntu. But under windows I always get the error “ImportError: cannot import name '_jpegpy' from partially initialized module 'lib.data_preprocess.utils.jpegpy' (most likely due to a circular import)”, this problem has been solved under Ubuntu by generating .so files, I would like to bother to ask why

**Comment by makaylacmac2** [comment]
I am experiencing many issues running this code on windows. The particular problem you mentioned I solved by modifying the /lib/data_preprocess/utils/jpegpy/jpegpy.py file like in the screenshot below. 

I am now working on handling a KeyError from the key './train_images\\manipulated_sequences\\Deepfakes\\183_253\\frame_0'. I modified the config file to begin with '.\\train_images.... as well as still receiving the same key error. 

![image](https://github.com/megvii-research/CADDM/assets/124900620/8f0ed8e8-d04d-48b1-9a0a-feb4670306d3)

**Comment by makaylacmac2** [comment]
@ez4encef Downgrading python in my environment to 3.10 got me past this issue at least. Now I am having a "No such file or directory" issue with the efficientnet-b3-5fb5a3c3.pth file mentioned in the latest issue posted.

**Comment by Ding-yd** [comment]
> 我在 Windows 上运行此代码时遇到了许多问题。您提到的特定问题我通过修改 /lib/data_preprocess/utils/jpegpy/jpegpy.py 文件解决了，如下面的屏幕截图所示。
> 
> 我现在正在努力处理来自密钥“./train_images\manipulated_sequences\Deepfakes\183_253\frame_0”的KeyError。我修改了配置文件以“.\train_images...”开头。以及仍然收到相同的密钥错误。
> 
> ![图像](https://private-user-images.githubusercontent.com/124900620/275361267-8f0ed8e8-d04d-48b1-9a0a-feb4670306d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NzI1MjQsIm5iZiI6MTcwNjc3MjIyNCwicGF0aCI6Ii8xMjQ5MDA2MjAvMjc1MzYxMjY3LThmMGVkOGU4LWQwNGQtNDhiMS05YTBhLWZlYjQ2NzAzMDZkMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwMVQwNzIzNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iODQ1M2Q5ZjAzYzE2YWYyZjY1NTE0ZTM4MGY0MTRlMTE0ZjgxOTE1MzUyODVmNDU3MjUyYTQxYmMyNGZiZDIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jWj_tHSibapmtLBtRC3rimPuQH2mB14yx9vNVG9790c)

Hi, I still get the same error after using your method

**Comment by wang-qichang** [comment]
I also get the same error

**Comment by BensonRen** [comment]
Thanks! 

> I am experiencing many issues running this code on windows. The particular problem you mentioned I solved by modifying the /lib/data_preprocess/utils/jpegpy/jpegpy.py file like in the screenshot below.
> 
> I am now working on handling a KeyError from the key './train_images\manipulated_sequences\Deepfakes\183_253\frame_0'. I modified the config file to begin with '.\train_images.... as well as still receiving the same key error.
> 
> ![image](https://private-user-images.githubusercontent.com/124900620/275361267-8f0ed8e8-d04d-48b1-9a0a-feb4670306d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Nzc0MTUsIm5iZiI6MTczMjU3NzExNSwicGF0aCI6Ii8xMjQ5MDA2MjAvMjc1MzYxMjY3LThmMGVkOGU4LWQwNGQtNDhiMS05YTBhLWZlYjQ2NzAzMDZkMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQyMzI1MTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNzAyNDM4MDQ1MGYwNzc0Y2M0ZWI1NTU1MTBiMzViM2FhODk5ZjczNmU3MmUwYzI5OTk0YjhmOWU4ODA3ZDE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1qBL3tE31520pGh3HBH1WaLxlOMWQnlTnheZ7vx3q7c)

Thank you so much! This one works!

**Comment by healer-hub** [comment]
> I am experiencing many issues running this code on windows. The particular problem you mentioned I solved by modifying the /lib/data_preprocess/utils/jpegpy/jpegpy.py file like in the screenshot below.
> 
> I am now working on handling a KeyError from the key './train_images\manipulated_sequences\Deepfakes\183_253\frame_0'. I modified the config file to begin with '.\train_images.... as well as still receiving the same key error.
> 
> ![image](https://private-user-images.githubusercontent.com/124900620/275361267-8f0ed8e8-d04d-48b1-9a0a-feb4670306d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4Mjc3ODcsIm5iZiI6MTczNTgyNzQ4NywicGF0aCI6Ii8xMjQ5MDA2MjAvMjc1MzYxMjY3LThmMGVkOGU4LWQwNGQtNDhiMS05YTBhLWZlYjQ2NzAzMDZkMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwMlQxNDE4MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMjM4ZmVjZTVhZmYwMDMyMDhhYzdhNDg3ZmIwNzFhZjU0MmYyMTAxMTNmMmI3ODQwM2YxM2ExZDcwYzRjZTlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bZ9mUY50jVtJJDM4_F5WMt6lKzLhojfw8n03iCoXj2k)

you are right men! here is something you need to do is delete ->from . import _jpegpy
